src/mapping/Partition.hpp

```diff
@@ -0,0 +1,315 @@
+#pragma once
+
+#include <Eigen/Core>
+
+#include <boost/container/flat_set.hpp>
+
+#include "mapping/RadialBasisFctSolver.hpp"
+#include "mapping/config/MappingConfiguration.hpp"
+#include "mapping/impl/BasisFunctions.hpp"
+#include "mesh/Filter.hpp"
+#include "precice/types.hpp"
+#include "utils/Event.hpp"
+
+namespace precice {
+
+namespace mapping {
+
+/**
+ * @brief Mapping using partition of unity decomposition strategies
+ */
+template <typename RADIAL_BASIS_FUNCTION_T>
+class Partition {
+public:
+  /**
+   * @brief Constructor.
+   *
+   * @param[in] dimension Dimensionality of the meshes
+   * @param[in] center Center of the particular partition
+   * @param[in] radius radius of the partition
+   * @param[in] parameter Shape parameter or support radius
+   * @param[in] targetSize target size (cardinality) of the partition
+   * @param[in] inputMesh the input mesh
+   * @param[in] outputMesh the output mesh
+   */
+  Partition(int               dimension,
+            mesh::Vertex      center,
+            double            radius,
+            double            parameter,
+            std::vector<bool> deadAxis,
+            Polynomial        polynomial,
+            unsigned int      targetSize,
+            mesh::PtrMesh     inputMesh,
+            mesh::PtrMesh     outputMesh);
+
+  /// Removes a computed mapping.
+  void clear();
+
+  /// Execute a consistent mapping
+  void mapConsistent(mesh::PtrData inputData, mesh::PtrData outputData);
+
+  /// Execute a consistent mapping
+  void mapConservative(mesh::PtrData inputData, mesh::PtrData outputData);
+
+  /// @brief set the normalized weight in the normalizedWeight data structure
+  void setNormalizedWeight(double normalizedWeight, VertexID vertexID);
+
+  /// Compute the weight for a given vertex
+  double computeWeight(const mesh::Vertex &v) const;
+
+  /// Evaluate the interpolant for a given vertex
+  double getInterpolatedValue(const VertexID id, unsigned int direction) const;
```

Comment:
```suggestion
  double getInterpolatedValue(const VertexID vertexID, unsigned int direction) const;
```
`dimension` or `axis` instead of `direction`?

Reply:
Function was obsolete.